fmlearlydisplay/src/main/java/net/minecraftforge/fml/earlydisplay/EarlyFramebuffer.java

```diff
@@ -0,0 +1,89 @@
+package net.minecraftforge.fml.earlydisplay;
+
+import java.nio.IntBuffer;
+
+import static org.lwjgl.opengl.GL32C.*;
+
+public class EarlyFramebuffer {
+    private final int framebuffer;
+    private final int texture;
+    private final FramebufferShader shader;
+    private final SimpleBufferBuilder bufferBuilder;
+
+    private RenderElement.DisplayContext context;
+
+    EarlyFramebuffer(final RenderElement.DisplayContext context, final FramebufferShader framebufferShader) {
+        this.context = context;
+        this.framebuffer = glGenFramebuffers();
+        this.texture = glGenTextures();
+        this.shader = framebufferShader;
+        glBindFramebuffer(GL_FRAMEBUFFER, this.framebuffer);
+        glActiveTexture(GL_TEXTURE0);
+        glBindTexture(GL_TEXTURE_2D, this.texture);
+        glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, context.width(), context.height(), 0, GL_RGBA, GL_UNSIGNED_BYTE, (IntBuffer)null);
+        glTexParameterIi(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+        glTexParameterIi(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+        glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, this.texture, 0);
+        glBindFramebuffer(GL_FRAMEBUFFER, 0);
+        this.bufferBuilder = new SimpleBufferBuilder(1);
+    }
+
+    void activate() {
+        glBindFramebuffer(GL_FRAMEBUFFER, this.framebuffer);
+    }
+
+    void deactivate() {
+        glBindFramebuffer(GL_FRAMEBUFFER, 0);
+    }
+
+    void draw(int width, int height) {
```

Comment:
Explain what this means please.

Reply:
The glBlitFrameBuffer function could be used instead of drawing a fullscreen quad after clearing the framebuffer. Behind the scenes its probably doing something very similar, but its easier to write/maintain.